core/src/main/java/kafka/server/share/SharePartitionManager.java

```diff
@@ -606,13 +616,15 @@ void maybeProcessFetchQueue() {
                 });
             });
 
-            Set<DelayedShareFetchKey> delayedShareFetchWatchKeys = new HashSet<>();
+            Set<Object> delayedShareFetchWatchKeys = new HashSet<>();
```

Comment:
Instead of using Object, perhaps we could have a base class or interface DelayedShareFetchKey and two specific subclasses like DelayedShareFetchGroupKey and DelayedShareFetchPartitionKey?